src/index.ts

```diff
@@ -125,14 +127,15 @@ export class PromiseQueue {
 					reject(e);
 				} finally {
 					this.metrics.emit('serviceTime', durationSince(serviceStartTime));
-					this.metrics.emit('latency', durationSince(enqueueTime));
-					this.metrics.emit('completion');
+					this.metrics.emit('latency', durationSince(arrivalTime));
+					this.metrics.emit('departure');
```

Comment:
Renaming is technically a breaking change I think but that said I'm not sure the reason for the name change as `completion` seems correct to me as the queued fn will have finished by this point due to the `await fn();` above

Reply:
You're right, not worth making a major version over a rename.